File: pegasusX/apps/driver-app-ios/driverappios/driverappios/Views/Components/Profile/HistorySection.swift
import SwiftUI

struct HistorySection: View {
    var vm: FleetViewModel

    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            HStack {
                Text("mobile_driver.ui.ride_history")
                    .font(.system(size: 17, weight: .bold))
                    .foregroundStyle(LabTheme.fg)

                Spacer()

                Text(L10n.format("mobile_driver.ui.count_rides", "\(vm.historyRows.count)"))
                    .font(.system(size: 12, weight: .medium, design: .monospaced))
                    .foregroundStyle(LabTheme.fgTertiary)
            }
            .padding(.horizontal, LabTheme.s8)

            if vm.historyRows.isEmpty {
                VStack(spacing: 10) {
                    Image(systemName: "clock.arrow.circlepath")
                        .font(.system(size: 24))
                        .foregroundStyle(LabTheme.fgTertiary)

                    Text("mobile_driver.ui.no_completed_rides_yet")
                        .font(.subheadline.weight(.medium))
                        .foregroundStyle(LabTheme.fgSecondary)
                }
                .frame(maxWidth: .infinity)
                .padding(.vertical, 30)
                .labCard()
            } else {
                ForEach(Array(vm.historyRows.enumerated()), id: \.element.id) { index, row in
                    HistoryRow(row: row, index: index)
                }
            }
        }
    }
}

struct HistoryRow: View {
    let row: DriverHistoryRow
    let index: Int

    var body: some View {
        HStack(spacing: 14) {
            ZStack {
                Circle()
                    .fill(LabTheme.success.opacity(0.15))
                    .frame(width: 36, height: 36)

                Image(systemName: "checkmark")
                    .font(.system(size: 13, weight: .bold))
                    .foregroundStyle(LabTheme.success)
            }

            VStack(alignment: .leading, spacing: 3) {
                Text(row.orderId)
                    .font(.system(size: 13, weight: .bold, design: .monospaced))
                    .foregroundStyle(LabTheme.fg)

                Text("\(row.status.isEmpty ? "COMPLETED" : row.status) · \(Int(row.totalMinor).formattedAmount)")
                    .font(.system(size: 11, weight: .medium))
                    .foregroundStyle(LabTheme.fgSecondary)
            }

            Spacer()

            StatusPill(label: row.status.isEmpty ? "DELIVERED" : row.status, color: LabTheme.success)
        }
        .padding(LabTheme.s16)
        .labCard()
        .staggeredAppear(index: index)
    }
}

File: pegasusX/apps/driver-app-android/app/src/main/java/com/pegasusx/driver/ui/screens/profile/components/HistorySection.kt
package com.pegasusx.driver.ui.screens.profile.components

import androidx.compose.ui.res.stringResource

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.CheckCircle
import androidx.compose.material.icons.filled.Schedule
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.pegasusx.driver.data.model.DriverHistoryRow
import com.pegasusx.driver.ui.components.PegasusCard
import com.pegasusx.driver.ui.components.StaggeredAppear
import com.pegasusx.driver.ui.components.StatusPill
import com.pegasusx.driver.ui.theme.LocalPegasusColors
import com.pegasusx.driver.ui.theme.PegasusSpacing
import com.pegasusx.driver.ui.theme.formattedAmount
import com.pegasusx.driver.R

@Composable
fun HistorySection(historyRows: List<DriverHistoryRow>) {
    val lab = LocalPegasusColors.current

    Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = PegasusSpacing.s8),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Text(
                text = stringResource(R.string.mobile_driver_ui_ride_history),
                fontSize = 17.sp,
                fontWeight = FontWeight.Bold,
                color = lab.fg,
            )
            Text(
                text = stringResource(R.string.mobile_driver_ui_size_rides, historyRows.size),
                fontSize = 12.sp,
                fontWeight = FontWeight.Medium,
                fontFamily = FontFamily.Monospace,
                color = lab.fgTertiary,
            )
        }

        if (historyRows.isEmpty()) {
            PegasusCard {
                Column(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(vertical = 30.dp),
                    horizontalAlignment = Alignment.CenterHorizontally,
                    verticalArrangement = Arrangement.spacedBy(10.dp)
                ) {
                    Icon(
                        imageVector = Icons.Default.Schedule,
                        contentDescription = null,
                        tint = lab.fgTertiary,
                        modifier = Modifier.size(24.dp)
                    )
                    Text(
                        text = stringResource(R.string.mobile_driver_ui_no_completed_rides_yet),
                        fontSize = 14.sp,
                        fontWeight = FontWeight.Medium,
                        color = lab.fgSecondary
                    )
                }
            }
        } else {
            historyRows.forEachIndexed { index, row ->
                StaggeredAppear(index = index) {
                    HistoryRow(row)
                }
            }
        }
    }
}

@Composable
fun HistoryRow(row: DriverHistoryRow) {
    val lab = LocalPegasusColors.current

    PegasusCard {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(PegasusSpacing.s16),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(14.dp)
        ) {
            Box(
                modifier = Modifier
                    .size(36.dp)
                    .clip(CircleShape)
                    .background(lab.success.copy(alpha = 0.15f)),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    imageVector = Icons.Default.CheckCircle,
                    contentDescription = null,
                    tint = lab.success,
                    modifier = Modifier.size(13.dp)
                )
            }

            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = row.orderId,
                    fontSize = 13.sp,
                    fontWeight = FontWeight.Bold,
                    fontFamily = FontFamily.Monospace,
                    color = lab.fg
                )
                Text(
                    text = "${row.status.ifBlank { "COMPLETED" }} · ${row.totalMinor.formattedAmount()}",
                    fontSize = 11.sp,
                    fontWeight = FontWeight.Medium,
                    color = lab.fgSecondary
                )
            }

            StatusPill(label = row.status.ifBlank { "DELIVERED" }, color = lab.success)
        }
    }
}
